main.py

```diff
@@ -31,16 +32,71 @@ def login():
     if not current_app.credit.check_user_with_password_exists(data['name'], data['password']):
         return 'Invalid name or password.', 403
         
-    return 'Correct name and password.', 200
+    return jsonify({'token': current_app.token_gen.create_new_token(data['name'], current_app.token_exp)}), 200
+
+def sign_in():
+    
+    try:
+        token = request.headers['Authorization']
+    except KeyError as inf:
+        logger.info(f'In post request Authorization header missing: {inf}')
+        return 'Please log in again', 401
     
+    token = token.replace('Bearer ', '')
+
+    if token == '':
+        logger.info('In Authorization header token missing')
+        return 'Please log in again', 401
+
+    if not current_app.token_gen.check_token(token):
+        logger.info('Got invalid token')
+        return 'Please log in again', 401
+
+    return jsonify({
+        'userName': 'denchik',
+        'id': 1,
+        'numOfLists': 1,
+        'toDoLists': {
+            'listName': 'First to-do list',
+            'listDate': '19 Dec 2019',
+            'listColor': 'green',
+            'listItems': [
+                {
+                    'itemNum': 1,
```

Comment:
Why would we send the `itermNum` here?

Reply:
`itemNum` represents the number of this list's item.